dll/win32/imm32/imm.c

```diff
@@ -1765,6 +1767,22 @@ HWND WINAPI ImmGetDefaultIMEWnd(HWND hWnd)
 UINT WINAPI ImmGetDescriptionA(
   HKL hKL, LPSTR lpszDescription, UINT uBufLen)
 {
+#ifdef __REACTOS__
+    IMEINFOEX info;
+    UINT cch;
+
+    TRACE("ImmGetDescriptionA(%p,%p,%d)\n", hKL, lpszDescription, uBufLen);
+
+    if (!ImmGetImeInfoEx(&info, ImeInfoExKeyboardLayout, &hKL) || !IS_IME_KBDLAYOUT(hKL))
+        return 0;
+
+    cch = (UINT)wcslen(info.wszImeDescription);
```

Comment:
Note: `wszImeDescription` is in a numbered buffer, so I don't know, whether it's possible it could be non-NULL-terminated (if too long).

Reply:
(In case of problems, `StringCchLengthW(info.wszImeDescription, _countof(info.wszImeDescription), &cch);`)

Reply:
OK.